packages/@aws-cdk/aws-config/lib/rule.ts

```diff
@@ -352,11 +353,20 @@ export class CustomRule extends RuleNew {
         messageType: 'ScheduledNotification',
       });
     }
-
-    props.lambdaFunction.addPermission('Permission', {
-      principal: new iam.ServicePrincipal('config.amazonaws.com'),
-      sourceAccount: this.env.account,
-    });
+    const hash = createHash('sha256')
+      .update(JSON.stringify({
+        fnName: props.lambdaFunction.functionName.toString,
+        accountId: Stack.of(this).resolve(this.env.account),
+        region: Stack.of(this).resolve(this.env.region),
+      }), 'utf8')
+      .digest('base64');
+    const customRulePermissionId: string = `CustomRulePermission${hash}`;
```

Comment:
Perhaps it is late in the day and I should not be reviewing, but is there a reason why the rules should not be part of the class attributes?
This looks like the `Create` part of CRUD -- is there no scenario by which management is required?